en/docs/insights/choreo-insights-api.md

```diff
@@ -0,0 +1,4217 @@
+<td valign="top"><a href="#int">Int</a></td>
+<td>
+
+Total number of rows in the result set. This value must be a positive integer. This field is optional. Hence, if
+there is no need of filtering by this attribute, omit the field. If limit value is not set, default limit value of
+5 will be used.
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>offset</strong></td>
+<td valign="top"><a href="#int">Int</a></td>
+<td>
+
+Offset value to set when filtering results. This value must be a positive integer. This field is optional. Hence, if
+there is no need of filtering by this attribute, omit the field. If offset value is not set, default offset value of
+0 will be used.
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>sortBy</strong></td>
+<td valign="top"><a href="#string">String</a></td>
+<td>
+
+Column name which will be used to sort results. This field is optional. Hence, if there is no need of filtering by
+this attribute, omit the field. If sortBy value is not set, default sortBy column name for the related operation
+will be used.
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>sortOrder</strong></td>
+<td valign="top"><a href="#string">String</a></td>
+<td>
+
+Order which will be used to sort results. The sortOrder value must be one of 'asc' or 'desc' values. This field is
+optional. Hence, if there is no need of filtering by this attribute, omit the field. If sortOrder value is not set,
+default sortOrder value of 'asc' will be used(with the exemption of setting 'desc' for count specific operations ex:
+getTopAPIsByAlertCount).
+
+</td>
+</tr>
+</tbody>
+</table>
+
+### ResourceUsageFilter
+
+Filters resource usage results.
+
+<table>
+<thead>
+<tr>
+<th colspan="2" align="left">Field</th>
+<th align="left">Type</th>
+<th align="left">Description</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td colspan="2" valign="top"><strong>paginationFilter</strong></td>
+<td valign="top"><a href="#paginationfilter">PaginationFilter</a></td>
+<td>
+
+The pagination filter. This is an optional parameter. Therefore, if there is no requirement to filter by this
+parameter, omit this parameter.
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>searchFilter</strong></td>
+<td valign="top"><a href="#searchfilter">SearchFilter</a></td>
+<td>
+
+The search filter. This is an optional parameter. Therefore, if there is no requirement to filter by this parameter,
+omit this parameter.
+
+</td>
+</tr>
+</tbody>
+</table>
+
+### SearchFilter
+
+Filters results by searching matching results for the provided search text.
+
+<table>
+<thead>
+<tr>
+<th colspan="2" align="left">Field</th>
+<th align="left">Type</th>
+<th align="left">Description</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td colspan="2" valign="top"><strong>apiIds</strong></td>
+<td valign="top">[<a href="#string">String</a>]</td>
+<td>
+
+List of API Ids where the 'searchtext' needs to be searched. This field is optional. Hence, if there is
```

Comment:
It is not clear how this works. Please check whether the following suggestion gives the correct meaning.

```suggestion
The list of APIs that need to be searched with the given search text. This is an optional parameter. Note that it is not possible to define API IDs without search text in the search filter.
```